jobs/ocp4/pds-ocp4-install.yaml

```diff
@@ -0,0 +1,258 @@
+                                ./oc create -f ./integreatly-operator/deploy/crds/examples/installation.cr.yaml -n ${namespaceName}
+                            """
+                        )
+                    } // stage
+
+                    stage ('Verify Installation Has Finished') {
+                        sh(
+                            returnStdout: false,
+                            script: """
+                                until ./oc get Installation example-installation -n ${namespaceName} --output=json -o jsonpath='{.status.stages.authentication.phase}' | grep -m 1 "completed"; do echo "Authentication stage not completed yet. Sleeping for 20 seconds";sleep 20 ; done
+
+                                until ./oc get Installation example-installation -n ${namespaceName} --output=json -o jsonpath='{.status.stages.products.phase}' | grep -m 1 "completed"; do echo "Products stage not completed yet. Sleeping for 20 seconds";sleep 20 ; done
+                            """
+                        )
+                    } // stage
+                } catch (caughtError){
+                    currentBuild.result = 'FAILURE'
+                    err = caughtError
+                    println caughtError
+                } finally {
+                    notifyBuild(currentBuild.result, err)
+                    
+                    if (cleanupAfterward) {
+                        cleanup(clusterAPI, namespaceName)
+                    }
+                }
+            }}} // timeout, ansiColor, timestamps
+        } // node
+
+        def notifyBuild(String buildStatus, err) {
+
+            // In theory, null means success. In practise if there is an error thrown elsewhere than in the triggered jobs (eg. in wokraround scripts)
+            // the job fails but success will be stored in buildStatus anyway
+            buildStatus =  buildStatus ?: 'SUCCESS'
+            
+            String mailRecipients = "${RECIPIENTS}"
+            String subject = "${buildStatus}: Job '${env.JOB_NAME} [${env.BUILD_NUMBER}]'"
+            String body = "${subject}\n\n(${env.BUILD_URL})"
+
+            if(err != null) {
+                body = "${body}\n\n FAILURE was ${err}"
+            }
+
+            mail body: body, subject: subject, to: mailRecipients
+        }
+
+        def cleanup(String clusterAPI, namespaceName) {
+            String cleanupOutput = sh(
+                returnStdout: true,
+                script: """
+                    ./oc delete namespace ${namespaceName} | true
+
+                    ./oc delete operatorsource integreatly-operators -n openshift-marketplace | true
+
+                    ./oc delete catalogsourceconfig installed-custom-${namespaceName} -n openshift-marketplace | true
+
+                    ./oc get crds | grep -v openshift.io | grep -v opentlc.com | grep -v k8s | grep -v coreos | grep -v automationbroker | grep -v NAME | awk '{print \$1}' | xargs ./oc delete crds | true
```

Comment:
yes, there are also oauthclients at the cluster scope that are created and should be deleted.

oc get oauthclients should show you what's left

Reply:
Cool, thanks. I will address this in separated PR.